src/firefly/java/edu/caltech/ipac/firefly/server/query/UserCatalogQuery.java

```diff
@@ -44,8 +48,22 @@ protected File loadDynDataFile(TableServerRequest req) throws IOException, DataA
 
         String filePath = req.getParam("filePath");
         if (!StringUtils.isEmpty(filePath)) {
-            File f = ServerContext.convertToFile(filePath);
-            return convertToIpacTable(f, req);
+            if (ServerParams.IS_WS.equals(req.getParam(ServerParams.SOURCE_FROM))) {
+                WsServerParams wsParams = new WsServerParams();
+                wsParams.set(WsServerParams.WS_SERVER_PARAMS.CURRENTRELPATH, filePath);
+                WsServerUtils wsUtil= new WsServerUtils();
+                try {
+                    String s=  wsUtil.upload(wsParams);
+                    return ServerContext.convertToFile(s);
+                } catch (IOException|FailedRequestException e) {
+                    throw new DataAccessException("Could now retrieve file from workspace",e);
+                }
+
+            }
+            else {
+                File f = ServerContext.convertToFile(filePath);
+                return convertToIpacTable(f, req);
```

Comment:
convertToIpacTable() should be called for files coming from workspace as well.  This is needed so we can upload fits, vo, or other formatted catalog tables.